spring-batch-infrastructure/src/main/java/org/springframework/batch/item/file/transform/DelimitedLineTokenizer.java

```diff
@@ -133,6 +134,8 @@ public void setQuoteCharacter(char quoteCharacter) {
 	@Override
 	protected List<String> doTokenize(String line) {
 
+		Assert.state(null != delimiter && 0 != delimiter.length());
```

Comment:
I agree.  It is not expected that the delimiter will change after a bean has been initialized.